stackslib/src/chainstate/nakamoto/signer_set.rs

```diff
@@ -445,4 +445,40 @@ impl NakamotoSigners {
         let name = Self::make_signers_db_name(reward_cycle, message_id);
         boot_code_id(&name, mainnet)
     }
+
+    /// Get the signer addresses and corresponding weights for a given reward cycle
+    pub fn get_signers_weights(
+        chainstate: &mut StacksChainState,
+        sortdb: &SortitionDB,
+        block_id: &StacksBlockId,
+        reward_cycle: u64,
+    ) -> Result<HashMap<StacksAddress, u64>, ChainstateError> {
```

Comment:
Is it an error of the signer set is empty?

Reply:
Good question. Not sure. But I can make it error here.

Reply:
Done.